pkg/generate/app/cmd/newapp.go

```diff
@@ -542,8 +543,36 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 						}
 					}
 				}
-				if pipeline, err = app.NewBuildPipeline(ref.Input().String(), input, c.OutputDocker, strategy, c.GetBuildEnvironment(environment), source); err != nil {
-					return nil, fmt.Errorf("can't build %q: %v", ref.Input(), err)
+
```

Comment:
It's illegal to specify --to when there are more than two builds defined - that check needs to be represented somewhere.